apps/web/app/(navigation)/MainMenu.tsx

```diff
@@ -0,0 +1,84 @@
+import * as Select from "@radix-ui/react-select";
+import styled from "styled-components";
+import { useRouter } from "next/navigation";
+import { signOut, useSession } from "next-auth/react";
+import { useState } from "react";
+
+import { useHasScrollBar } from "lib/hooks/useHasScrollBar";
+import { Avatar, Icon } from "ui";
+import { useTranslate } from "lib/hooks";
+
+import {
+  SelectContentStyled,
+  SelectItemStyled,
+  SelectTriggerStyled,
+} from "./LanguageSelectorStyled";
+
+export const AvatarStyled = styled(Avatar)`
+  height: 2.1em;
+  width: 2.1em;
+`;
+
+const IconUpStyled = styled(Icon)`
+  transform: rotate(180deg);
+`;
+
+export const MainMenu = () => {
+  const { hasScrollbar } = useHasScrollBar();
+  const router = useRouter();
+  const { data } = useSession();
+  const { t, dict } = useTranslate("MainPage");
+
+  const [isIconDown, setIsIconDown] = useState<boolean | undefined>();
+
+  const IconUp = () => <IconUpStyled icon="arrow_drop_down" />;
+  const IconDown = () => <Icon icon="arrow_drop_down" />;
+
+  const menuHandler = (value: string) => {
+    value === "sign-out" ? signOut() : router.push("/");
+  };
+
+  const items = [
+    {
+      value: "home",
+      text: t(dict.home),
+    },
+    {
+      value: "sign-out",
+      text: t(dict.signOut),
+    },
+  ];
+
+  return (
+    <>
+      {data && (
+        <Select.Root
+          onValueChange={(value) => menuHandler(value)}
```

Comment:
I am pointing that one out again ;-) `menuHandler` is already a function that takes `value`

```suggestion
          onValueChange={menuHandler}
```
but I worked out that there is an issue anyway. With the way you have it right now, redirecting to `home` works only the first time. After that, `select` "remembers" the previously chosen value and does not call `onValueChange`.

check this with the current code that you have: 
- login
- stay on the home page or go the budget page
- select `home` from your select
- page reloads correctly
- go to other budget page or stay on home and select home again - page does not reload anymore

How to fix it? add `value=""` in `Select.Root`
```
        <Select.Root
          value=""
          onValueChange={menuHandler}
```
... and please include the suggested change, since it is really the same :)

Reply:
Oooo! Youre right, that is wierd. I'll fix it.